x/farming/keeper/proposal_handler.go

```diff
@@ -143,22 +150,32 @@ func (k Keeper) UpdatePublicPlanProposal(ctx sdk.Context, proposals []*types.Upd
 				}
 			}
 
-			if p.GetName() != "" {
-				plan.Name = p.GetName()
-			}
-
-			if p.GetEpochAmount() != nil {
-				plan.EpochAmount = p.GetEpochAmount()
+			// change the plan to fixed amount plan if an epoch amount exists
+			if p.GetEpochAmount().IsAllPositive() {
```

Comment:
It's not necessary but for simplicity, we could add `GetBasePlan` method to the `PlanI` interface and use that here as:
```go
plan = types.NewFixedAmountPlan(plan.GetBasePlan(), p.GetEpochAmount())
```

Reply:
Yes, I agree that makes it simpler. Let me make that change.